src/chain/chain.rs

```diff
@@ -1071,11 +1055,11 @@ impl Chain {
             .into_iter()
             .flat_map(|(_, si)| {
                 si.member_names()
-                    .into_iter()
                     .sorted_by(|name0, name1| name.cmp_distance(name0, name1))
             })
             .filter(|name| connected_peers.contains(&name))
             .take(count)
+            .cloned()
```

Comment:
You can use `copied` here as `XorName` implements `Copy`. In general it's a good idea to use `copied` instead of `cloned` so that if we ever make `XorName` non-`Copy` for whatever reason, this code would stop compiling which would make us aware of a potential performance impact.

Reply:
Yep agree, makes sense!